### PR TITLE
update riak version to 2.2.3 and update download location

### PR DESCRIPTION
**Pull request by randysecrist**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by randysecrist** [comment]
Basho (the company that made and hosted the binary file before) has gone under.  A separate group has decided to host the files needed to keep things working.

**Comment by DomT4** [comment]
Has the new upstream been "blessed" by someone with some knowledge of the situation? It seems highly questionable to simply trust a new source unless that's the case.

I found [this email](http://lists.basho.com/pipermail/riak-users_lists.basho.com/2017-August/019500.html) but nothing about the new download URL.

File: Formula/riak.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Riak < Formula
   desc "Distributed database"
   homepage "http://basho.com/products/riak-kv/"
-  url "https://s3.amazonaws.com/downloads.basho.com/riak/2.2/2.2.2/osx/10.8/riak-2.2.2-OSX-x86_64.tar.gz"
-  version "2.2.2"
-  sha256 "73fef949afe3864559fd4ead4cb368749259fac2162a9c931bed20d131f5b8cb"
+  url "https://files-source.tiot.jp/riak/kv/2.2/2.2.3/osx/10.8/riak-2.2.3-OSX-x86_64.tar.gz"
```
Comment (ilovezfs) [inline Formula/riak.rb]: do you have a reference for this change?

**Review by ilovezfs — CHANGES_REQUESTED** [review]

**Comment by ilovezfs** [comment]
https://files-source.tiot.jp/riak/kv/2.2/2.2.2/osx/10.8/riak-2.2.2-OSX-x86_64.tar.gz actually has the same checksum as is in the formula for 2.2.2 right now, so that's another option.

File: Formula/riak.rb (hunk 1 above)
Comment (randysecrist) [reply Formula/riak.rb]: I have this URL from the basho repo; and from discussion with some former engineers.  I've considered hosting another public mirror myself as well.

https://github.com/basho/riak/issues/922

2.2.3 was the last release ; I don't think there will be more releases or direction > 2.2.3 until after the court stuff is done.  This PR is to simply keep things functional until that happens.

**Comment by randysecrist** [comment]
Since the checksum is the same with 2.2.2; I'm ok with keeping things there instead of moving to 2.2.3.  In think the more important change is having a URL that works.

Whatever the brew folks decide is the best policy for you all is ok with me.

**Comment by ilovezfs** [comment]
@randysecrist I'm working on a source build of riak :)

**Comment by randysecrist** [comment]
👍 to source builds.  :)

It just gets a bit messy since the code is still wired up to a now very old build of erlang; there are branches in places where things work against OTP 19, 20 etc.

**Comment by ilovezfs** [comment]
>It just gets a bit messy since the code is still wired up to a now very old build of erlang; there are branches in places where things work against OTP 19, 20 etc.

I'm going to add an erlang@17 formula.

**Comment by ilovezfs** [comment]
@randysecrist one issue I'm hitting is that it doesn't want to print the version when I run `riak version`

**Comment by ilovezfs** [comment]
```
iMac-TMP:homebrew-core joe$ riak version

iMac-TMP:homebrew-core joe$
```

**Comment by ilovezfs** [comment]
@randysecrist it seems to just be a bug in the `make rel` Makefile target. If I use `make package` then it works.

**Comment by randysecrist** [comment]
The "rel" target is for internal engineering work and "package" for releases so makes sense.

**Comment by ilovezfs** [comment]
@randysecrist I've shipped https://github.com/Homebrew/homebrew-core/pull/17292, which builds 2.2.3 from source. Thanks for the heads up about the situation :)

**Comment by koudelka** [comment]
Hey @ilovezfs, have you tried running riak 2.2.3 from a fresh homebrew install? I've tried both the bottle and source builds, it crashes with barely a peep to stdout/err and logs.

I've tried it on three different machines with varying configurations, all OS X 10.8.

```
❯ riak console
config is OK
-config /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/app.2017.09.12.17.11.43.config -args_file /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.12.17.11.43.args -vm_args /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.12.17.11.43.args
!!!!
!!!! WARNING: ulimit -n is 256; 65536 is the recommended minimum.
!!!!
Exec:  /usr/local/Cellar/riak/2.2.3/libexec/erts-6.4.1.6/bin/erlexec -boot /usr/local/Cellar/riak/2.2.3/libexec/releases/2.2.3/riak               -config /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/app.2017.09.12.17.11.43.config -args_file /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.12.17.11.43.args -vm_args /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.12.17.11.43.args              -pa /usr/local/Cellar/riak/2.2.3/libexec/lib/basho-patches -- console
Root: /usr/local/Cellar/riak/2.2.3/libexec
Erlang/OTP 17 [erts-6.4.1.6] [source] [64-bit] [smp:8:8] [async-threads:64] [hipe] [kernel-poll:true] [dtrace]

[1]    35406 abort      riak console
```

```
❯ cat /usr/local/var/log/riak/error.log

~
❯ cat /usr/local/var/log/riak/crash.log

~
❯ cat /usr/local/var/log/riak/console.log
2017-09-12 17:12:59.430 [info] <0.7.0> Application lager started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.441 [info] <0.7.0> Application sasl started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.441 [info] <0.7.0> Application asn1 started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.441 [info] <0.7.0> Application crypto started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.441 [info] <0.7.0> Application public_key started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.450 [info] <0.7.0> Application ssl started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.454 [info] <0.7.0> Application riak_sysmon started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.464 [info] <0.7.0> Application os_mon started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.471 [info] <0.7.0> Application runtime_tools started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.474 [info] <0.75.0> alarm_handler: {set,{{disk_almost_full,"/"},[]}}
2017-09-12 17:12:59.478 [info] <0.75.0> alarm_handler: {set,{system_memory_high_watermark,[]}}
2017-09-12 17:12:59.506 [info] <0.7.0> Application erlang_js started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.506 [info] <0.7.0> Application xmerl started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.523 [info] <0.7.0> Application inets started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.523 [info] <0.7.0> Application mochiweb started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.530 [info] <0.7.0> Application webmachine started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.530 [info] <0.7.0> Application basho_stats started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.537 [info] <0.7.0> Application bitcask started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.552 [info] <0.7.0> Application clique started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.552 [info] <0.7.0> Application eleveldb started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.552 [info] <0.7.0> Application pbkdf2 started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.552 [info] <0.7.0> Application poolboy started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.580 [info] <0.158.0>@exometer_report:do_start_reporters:613 Starting reporters with []
2017-09-12 17:12:59.580 [info] <0.7.0> Application exometer_core started on node 'riak@127.0.0.1'
2017-09-12 17:12:59.661 [warning] <0.178.0>@riak_core_ring_manager:reload_ring:359 No ring file available.
2017-09-12 17:12:59.853 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,vnode_routing} = proxy
2017-09-12 17:12:59.871 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,staged_joins} = true
2017-09-12 17:12:59.890 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,resizable_ring} = true
2017-09-12 17:12:59.904 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,fold_req_version} = v2
2017-09-12 17:12:59.922 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,security} = true
2017-09-12 17:12:59.937 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,bucket_types} = true
2017-09-12 17:12:59.952 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_core,net_ticktime} = true
2017-09-12 17:13:00.083 [info] <0.7.0> Application riak_core started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.094 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_pipe,trace_format} = ordsets
2017-09-12 17:13:00.146 [info] <0.7.0> Application riak_pipe started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.152 [info] <0.7.0> Application sidejob started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.152 [info] <0.7.0> Application riak_dt started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.152 [info] <0.7.0> Application protobuffs started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.152 [info] <0.7.0> Application riak_pb started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.205 [info] <0.7.0> Application riak_api started on node 'riak@127.0.0.1'
2017-09-12 17:13:00.270 [info] <0.333.0>@riak_kv_env:doc_env:46 Environment and OS variables:
2017-09-12 17:13:00.315 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_kv,object_hash_version} = 0
2017-09-12 17:13:00.332 [info] <0.198.0>@riak_core_capability:process_capability_changes:555 New capability: {riak_kv,anti_entropy} = enabled_v1
2017-09-12 17:13:00.349 [info] <0.374.0>@riak_core_throttle:enable_throttle:119 Enabling throttle for riak_kv/aae_throttle.
2017-09-12 17:13:00.414 [info] <0.381.0>@riak_core:wait_for_service:504 Waiting for service riak_kv to start (0 seconds)
```

It looks like the VM terminates without leaving a crash dump. Watching it with DTrace didn't turn up much for me either.

Any thoughts?

**Comment by ilovezfs** [comment]
>WARNING: ulimit -n is 256; 65536 is the recommended minimum.

Did you address that?

**Comment by koudelka** [comment]
@ilovezfs i did, yes. I've had it functioning properly for a few years now with `ulimit -n 9999`.

**Comment by ilovezfs** [comment]
No idea, then.

**Comment by koudelka** [comment]
@ilovezfs well, thanks. i guess we'll see if others report it too.

**Comment by jpmoss** [comment]
I'm seeing the exact same thing here, macOS 10.12.

**Comment by randysecrist** [comment]
Could be related to using erlang 17; all releases use bashos own distribution.

Might be better to switch it to use a built binary from a mirror until bet365 provides updates

Typed on tiny keys

**Comment by ilovezfs** [comment]
There have been 408 installs in the last 21 days, and this is the only report of breakage. So I'll need to see more evidence something is broken before changing anything. Unfortunately, the next step here would be to remove riak from Homebrew, not to reinstate a binary version, which would belong in Cask.

**Comment by koudelka** [comment]
Fair enough. For what it's worth, I think you'll be able to reproduce this on a fresh homebrew install, if you'd like to see if it's actually broken.

**Comment by ilovezfs** [comment]
```
iMac-TMP:~ joe$ riak console
config is OK
-config /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/app.2017.09.18.10.02.11.config -args_file /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.18.10.02.11.args -vm_args /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.18.10.02.11.args
!!!!
!!!! WARNING: ulimit -n is 256; 65536 is the recommended minimum.
!!!!
Exec:  /usr/local/Cellar/riak/2.2.3/libexec/erts-6.4.1.6/bin/erlexec -boot /usr/local/Cellar/riak/2.2.3/libexec/releases/2.2.3/riak               -config /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/app.2017.09.18.10.02.11.config -args_file /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.18.10.02.11.args -vm_args /usr/local/Cellar/riak/2.2.3/libexec/data/generated.configs/vm.2017.09.18.10.02.11.args              -pa /usr/local/Cellar/riak/2.2.3/libexec/lib/basho-patches -- console
Root: /usr/local/Cellar/riak/2.2.3/libexec
Erlang/OTP 17 [erts-6.4.1.6] [source] [64-bit] [smp:8:8] [async-threads:64] [hipe] [kernel-poll:true] [dtrace]

{"Kernel pid terminated",application_controller,"{application_terminated,os_mon,shutdown}"}

Crash dump was written to: ./log/erl_crash.dump
Kernel pid terminated (application_controller) ({application_terminated,os_mon,shutdown})
iMac-TMP:~ joe$ 
```
is what I see.

**Comment by koudelka** [comment]
That may have happened because you hit the open files limit (as opposed to the issue above), the log files will indicate if so.

If you increase your `ulimit -n`, you should see the same output as above.

**Pull request closed without merging**